benchto-driver/src/main/java/io/trino/benchto/driver/execution/QueryExecutionDriver.java

```diff
@@ -61,12 +79,35 @@ private boolean isSelectQuery(String sql)
         return sql.startsWith("select") || sql.startsWith("show") || sql.startsWith("with");
     }
 
-    private QueryExecutionResult executeSelectQuery(Connection connection, QueryExecutionResultBuilder queryExecutionResultBuilder, String sqlStatement)
+    private QueryExecutionResult executeSelectQuery(
+            Connection connection,
+            QueryExecutionResultBuilder queryExecutionResultBuilder,
+            String sqlStatement,
+            Optional<Path> resultFile)
             throws SQLException
     {
         try (Statement statement = connection.createStatement();
                 ResultSet resultSet = statement.executeQuery(sqlStatement)) {
-            int rowsCount = fetchRows(sqlStatement, resultSet);
+            int rowsCount;
+            if (resultFile.isPresent()) {
+                // use results file
+                if (Files.exists(resultFile.get())) {
+                    // load results from file and compare
+                    rowsCount = compareRows(resultFile.get(), resultSet);
+                }
+                else {
+                    if (properties.requireResults()) {
+                        // we require results file to exist
+                        throw new RuntimeException("Required result file not found: " + resultFile.get());
+                    }
+                    // dump results to file
```

Comment:
We we do we dump results? How this will be used?

Shouldn't we pre-compute results beforehand. If we generate and compare results using faulty Trino, then such assertion is not really that much valuable.